activities.py

```diff
@@ -0,0 +1,69 @@
+# @@@SNIPSTART python-money-transfer-project-template-withdraw
+import asyncio
+
+from temporalio import activity
+
+from banking_service import BankingService, InvalidAccountError
+from shared import PaymentDetails
+
+
+class BankingActivities:
+    def __init__(self):
+        self.bank = BankingService("bank-api.example.com")
+
+    @activity.defn
+    async def withdraw(self, data: PaymentDetails) -> str:
+        reference_id = f"{data.reference_id}-withdrawal"
+        try:
+            confirmation = await asyncio.to_thread(
+                self.bank.withdraw, data.source_account, data.amount, reference_id
+            )
+            return confirmation
+        except InvalidAccountError:
+            raise
+        except Exception as error:
+            activity.logger.exception("Withdrawal failed")
+            raise error
```

Comment:
```suggestion
        except Exception:
            activity.logger.exception("Withdrawal failed")
            raise
```

(here and elsewhere)

Reply:
Fixed.